### ISSUE TITLE
feature request: deniable aliases

### ISSUE THREAD

**Issue by ghostlands** [irungentoo/toxcore#1552]
Just to be clear, the idea being to have one identity with several addresses that cannot be proven to derive from one another.

The why is that it takes a lot of bandwidth currently to keep several clients open all the time when the purpose is to be available via several identities. Without any kind of offline messages (just to be clear, I'm not asking for such), keeping in touch via more than one identity is an intensive, expensive, and manual task.

Apologies if this suggestion would take just as much bandwidth but in a different way. Putting the idea out there in case this isn't necessarily so.

**Comment by GrayHatter** [comment]
<irungentoo> that can't work. If you do come up with something more workable, or a better/more detailed suggestion. You can open a new issue

**Comment by ghostlands** [comment]
Ok, thanks. I'll continue to think about it.

**Comment by ghostlands** [comment]
@GrayHatter

Not to be a pest, but I remembered why I thought this idea might be possible. In BitMessage, the deterministic address option allows the user to generate multiple addresses: https://bitmessage.org/wiki/Address#Deterministic_Address

I don't recall if those adresses were deniable, I may just have assumed so at the time, because it was an obvious necessity (it was a long years ago that I last used BitMessage).

Obviously deterministic addresses require a lot of hand-holding on the part of the client to ensure that users input elaborate passphrases, so I'm not necessarily advocating deterministic addresses.

BitMessage is of course here on GitHub: https://github.com/Bitmessage/PyBitmessage

Anyway, screenshots below.

**Comment by GrayHatter** [comment]
@ghostlands you're not a bother at all. A well formatted message with plenty of information and sources? I wish all messages were this good.

I'm going to reopen this because now you have something workable. 

Here's a few problems that I see, that maybe you can come up with a solution for: How to answer from a deniable alias, while still being findable?

You're right you can have multiple addresses/ToxIDs and with it the huge overhead of multiple instances. But even if we give you X number of alias, how do you find your contact? How do you maintain an ability to deny the existence/use of that alias while still being able to prove that it only could have come from you?

Finally, what are you directly trying to gain from this solution? Maybe there's another way, or maybe the feature already exists. 

I don't mean to seem harsh, just a few issues that need to be resolved before we can add anything like this to Tox.

**Comment by ghostlands** [comment]
I may be at my current limit for helping, but here are my current best responses anyway:

> How to answer from a deniable alias, while still being findable?

> how do you find your contact?

Reading your whole entry, I think we might be thinking different things re: deniability. I only want it to be impossible to determine mathematically if addresses are derived from the same seed. Methods to correlate them using other methods such as timing and other network activity are later contingencies outside the scope of my bug as intended. Basically unless I'm misunderstanding you, I don't really get the question, because aliases in my thinking should be deployed and responded to/with no differently than normal current addresses; the only difference being they could all be online in the same client at the same time.

> How do you maintain an ability to deny the existence/use of that alias while still being able to prove that it only could have come from you?

This is the part where I think we might be thinking different things about the word "deniable". Whoever you exchange that alias address with should be able to trust it every bit as well as any address that is the current norm. They just shouldn't be able to see that it has any relation to any other alias you have out there, should they somehow gather a range of addresses for comparison.

No harshness on your part assumed by me. These are technical issues and careful planning and grasp are required.

One additional thought that is just outside my scope, but seems important, is beyond the timing correlation problem. Currently, without Tor as a proxy, are the user's IPs transparent? If so, correlation between aliases would happen first there, and the whole thing would be useless (except as a spam deterrent, which Tox already has a solution for).

**Comment by ghostlands** [comment]
Another side thought is that if only as a spam deterrent, and if deniability were not important (to be clear, I do think it's crucial) I wonder if the current nospam feature might be extended to become non-deniable multiple addresses/aliases. In this case addresses would explicitly be corollary under analysis and would only be used for convenience of superficially organizing contacts. Average individuals offered one of your "nospam" aliases, given an example of your primary address, could determine the addresses were connected, but would have a difficult time guessing your other aliases. Clients would have the option of setting different notification options and privileges for them.

**Comment by ghostlands** [comment]
I know nospam already _sort of_ can be a non-deniable alias. But it doesn't look like - at least on the client-side - the user has the option to re-use a previous version of their address once a new nospam has been generated.

**Comment by GrayHatter** [comment]
nospam is an entirely different concept, so I agree that it's not applicable in this case.

If I'm understanding you correctly, you don't want deniable aliases you just want a separate ID/key for different friends/groups. Such that: if you give address_a to you friend bob, and address_b to your friend frank, even if bob and frank are the same person, they'd have no way to know/prove that address_a and address_b are both you?

**Comment by ghostlands** [comment]
That's a pretty unambiguously correct interpretation. Sorry if I was using confusing terminology.

I'm hoping this is what Bitmessage already had implemented.
